apps/web/src/design-system/table/Table.tsx

```diff
@@ -91,10 +93,8 @@ export function Table<T extends object>({
   ) as unknown as UseTableProps<T>;
 
   useEffect(() => {
-    if (onPageChange) {
-      onPageChange(pageIndex);
-    }
-  }, [pageIndex]);
+    onPageChangeRef.current?.(pageIndex);
```

Comment:
the `onPageChange` property changes on every render, so to not break this `useEffect` I do pass it through the ref

Reply:
Would not it be easier to use useCallback?

Reply:
no, because it will be re-created on every render, because `onPageChange` prop changes every render... but I'll think about this as I think we should be able to remove `useEffect` here at all